2420/2/options_file.rb

```diff
@@ -0,0 +1,60 @@
+require 'optparse'
+require './versus.rb'
+require 'terminal-table'
+
+options = { name: '', top_bad_words: 0, top_words: 30 }
+OptionParser.new do |opts|
+  opts.banner = 'Usage: example.rb [options]'
+
+  opts.on('--top-bad-words [NUMBER]', Integer, 'Максимальное количество самых нецензурных участников') do |number|
+    options[:top_bad_words] = number
+  end
+  opts.on('--name [NAME]', String, 'Имя участника') do |name|
+    options[:name] = name
+  end
+  opts.on('--top-words [NUMBER]', Integer, 'Любимые слова человека') do |number|
+    options[:top_words] = number || 30
+  end
+end.parse!
+
+if options[:top_bad_words] > 0
+  battlers = Battler.battler_names_list
+                    .map { |name| Battler.new(name) }
+                    .sort_by(&:bad_words_count)
+                    .pop(options[:top_bad_words])
+                    .reverse
+  rows = []
+  battlers.map do |member|
+    rows << [member.name.to_s,
+             "#{member.battles_count} батлов",
+             "#{member.bad_words_count} нецензурных слов ",
+             " #{member.words_in_battle}  слова на батл ",
+             " #{member.words_in_raund} слова в раунде"]
+  end
+  table = Terminal::Table.new rows: rows
+  puts table
+end
+
+unless options[:name].empty?
+  rows = []
+  name_battler = []
+  names = Battler.battler_names_list
+  names.each_with_index { |name, index| name_battler[index] = name.split.first }
```

Comment:
you don't need index here, you can use simple each and `name_battler << name.split.first`

also, why do you need to `split.first` on name?

Reply:
each_whith_index - it's you say me to fix

split first - because i look for one part name_battlers
not " Billy Milligan", but "billy"

Reply:
You don't need an index here. https://apidock.com/ruby/Array/each